docs/decisions/0002-authorization-model-foundation.rst

```diff
@@ -0,0 +1,201 @@
+* We express authorization as: is **Subject** allowed to do **Action** on **Object** under **Context**?
+* This normalization is used in policies, code, queries, and audits.
+* Examples:
+
+  - Can Alice (subject) edit (action) the course ``course-v1:OpenedX+DemoX+DemoCourse`` (object) as part of Org A (context)?
+  - Can Bob (subject) read (action) the library ``lib:DemoX:CSPROB`` (object)?
+
+II. Resources and Scopes
+========================
+
+Scopes as first-class citizens in permission-granting
+-----------------------------------------------------
+* A **scope** defines the boundary within which a role or policy applies (for example: platform-wide, organization-wide, a single course, or a specific library).
+* Treating scopes as **first-class citizens** means they are explicitly modeled in the system, not hidden inside ad-hoc resource definitions. They must be available to policies, queries, and audits in a consistent way.
+* Scopes can be **parameterized** (e.g., ``organization:ORG-A``, ``course:course-v1:OpenedX+DemoX+DemoCourse``,  ``site:sandbox.openedx.org``, ``instance``) to support granular checks.
+* **Inheritance across scopes** must be supported (e.g., permissions granted at the organization level can cascade to courses in that organization when intended).
+* By making scopes explicit and consistent, we avoid the fragmentation seen in legacy systems (different services using different implicit notions of "instance", "org", "course").
+* Scope is part of the **Context** in S-A-O-C checks.
+
+III. Authorization Paradigm
+===========================
+
+Adopt ABAC as the goal; Scoped RBAC as a first step
+---------------------------------------------------
+* We recommend **ABAC** as the main model for Open edX authorization.
+* **Scoped RBAC** may be used pragmatically as a first step, with the ambition of moving into a more granular system with ABAC.
+* **RBAC** handles role-based permissions well (e.g., "admins can edit any record").
+* **ABAC** adds finer control by using attributes of subjects, resources, and context (e.g., "editors can edit only in their assigned organizations or locations").
+* **ReBAC** is not chosen because it adds complexity and we do not have strong use cases today.
+
+  - Although ReBAC solves interesting problems out of the box (inheritance, recursive relationships), it introduces a mental shift in how to think about authorization so we're not explicitly adopting it for now.
+  - Some technologies are ReBAC-first but can also implement RBAC and ABAC effectively. These are not excluded, but they shouldn't go against our **simplicity principle**.
+
+* **Simplicity principle**: avoid adding features like deep role inheritance or complex hierarchies until there are clear use cases that require them.
+
+IV. Policy Definition
+=====================
+
+Externalize policies
+--------------------
+* Policies must be defined outside code, not hardcoded with conditionals.
```

Comment:
Do we want to codify here that the policy is an allowlist and we're not supporting explicit denial (at least in the default policy, as I understand it)?